app/code/Magento/GroupedProduct/Model/Quote/Item/CartItemProcessor.php

```diff
@@ -0,0 +1,127 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\GroupedProduct\Model\Quote\Item;
+
+use Magento\Framework\DataObject;
+use Magento\Framework\DataObject\Factory as ObjectFactory;
+use Magento\Framework\Serialize\Serializer\Json;
+use Magento\GroupedProduct\Api\Data\GroupedOptionsInterface;
+use Magento\GroupedProduct\Api\Data\GroupedOptionsInterfaceFactory;
+use Magento\GroupedProduct\Model\Product\Type\Grouped;
+use Magento\Quote\Api\Data as QuoteApi;
+use Magento\Quote\Api\Data\CartItemInterface;
+use Magento\Quote\Model\Quote\Item\CartItemProcessorInterface;
+
+/**
+ * Converts grouped_options to super_group for the grouped product.
+ */
+class CartItemProcessor implements CartItemProcessorInterface
+{
+    private const SUPER_GROUP_CODE = 'super_group';
+
+    /**
+     * @var ObjectFactory
+     */
+    private $objectFactory;
+
+    /**
+     * @var GroupedOptionsInterface
+     */
+    private $groupedOptionFactory;
+
+    /**
+     * @var Json
+     */
+    private $jsonSerializer;
+
+    /**
+     * @var QuoteApi\ProductOptionExtensionFactory
+     */
+    private $productOptionExtensionFactory;
+
+    /**
+     * @var QuoteApi\ProductOptionInterfaceFactory
+     */
+    private $productOptionFactory;
+
+    /**
+     * @var array|null
+     */
+    private $groupedOptions;
+
+    /**
+     * @param ObjectFactory $objectFactory
+     * @param GroupedOptionsInterfaceFactory $groupedOptionFactory
+     * @param Json $jsonSerializer
+     * @param QuoteApi\ProductOptionExtensionFactory $productOptionExtensionFactory
+     * @param QuoteApi\ProductOptionInterfaceFactory $productOptionFactory
+     */
+    public function __construct(
+        ObjectFactory $objectFactory,
+        GroupedOptionsInterfaceFactory $groupedOptionFactory,
+        Json $jsonSerializer,
+        QuoteApi\ProductOptionExtensionFactory $productOptionExtensionFactory,
+        QuoteApi\ProductOptionInterfaceFactory $productOptionFactory
+    ) {
+        $this->objectFactory = $objectFactory;
+        $this->groupedOptionFactory = $groupedOptionFactory;
+        $this->jsonSerializer = $jsonSerializer;
+        $this->productOptionExtensionFactory = $productOptionExtensionFactory;
+        $this->productOptionFactory = $productOptionFactory;
+    }
+
+    /**
+     * Converts the grouped_options request data into the same format as native frontend add-to-cart
+     *
+     * @param CartItemInterface $cartItem
+     * @return DataObject|null
+     */
+    public function convertToBuyRequest(CartItemInterface $cartItem): ?DataObject
+    {
+        $extensionAttributes = $cartItem->getProductOption()
+            ? $cartItem->getProductOption()->getExtensionAttributes()
+            : null;
+        if ($extensionAttributes) {
```

Comment:
I guess a negative here and return null it would make the code more readable as it will reduce one level on the chain.